### ISSUE TITLE
Some functionalities broken after update to 5.9.0

### ISSUE THREAD

**Issue by setzer22** [ionide/ionide-vscode-fsharp#1613]
### Describe the bug
After an automatic update to 5.9.0 this morning, most of my editor features stopped working. I am experiencing the exact same behavior as reported by this user: https://github.com/ionide/ionide-vscode-fsharp/issues/1173#issuecomment-599375520

Downgrading to yesterday's 5.8.1 resolved the issue.

### Steps to reproduce
1. Update to 5.9.0 (or have VSCode automatically update it)
2. Issue starts occurring

### Expected behaviour
Functionality like renaming or changing project order works as expected.

## Machine info
 - OS: Linux
 - .NET SDK version: 5.0.205
 - Ionide version: 5.9.0

**Comment by witoldsz** [comment]
My environment fails to launch the server after upgrade to 5.9.0:
```
The F# server crashed 5 times in the last 3 minutes. The server will not be restarted.
```

```
[Info  - 3:40:09 PM] Connection to server got closed. Server will restart.
[Info  - 3:40:09 PM] Connection to server got closed. Server will restart.
[Info  - 3:40:10 PM] Connection to server got closed. Server will restart.
[Info  - 3:40:10 PM] Connection to server got closed. Server will restart.
[Error - 3:40:11 PM] Connection to server got closed. Server will not be restarted.
```

```
$ dotnet --list-sdks
5.0.403 [/usr/share/dotnet/sdk]

$ dotnet --list-runtimes
Microsoft.AspNetCore.App 5.0.12 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 5.0.12 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

$ uname -a
Linux home 5.11.0-40-generic #44~20.04.2-Ubuntu SMP Tue Oct 26 18:07:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
Nothing but the syntax coloring works, downgrading the Ionide-fsharp plugin to 5.8.1 solves the issue.

**Comment by cartermp** [comment]
@witoldsz what happens if you have .NET 6 installed? Just released yesterday, but it's an LTS release.

**Comment by witoldsz** [comment]
OK, so I have installed `dotnet-sdk-6.0` from repositories for Ubuntu.
```
$ dotnet --version
6.0.100

$ dotnet --list-sdks
5.0.403 [/usr/share/dotnet/sdk]
6.0.100 [/usr/share/dotnet/sdk]

$ dotnet --list-runtimes
Microsoft.AspNetCore.App 5.0.12 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 5.0.12 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

But nothing's changed, i.e. the F# plugin does not work, the sever won't run.

**Comment by Krzysztof-Cieslak** [comment]
@witoldsz Please check log in the F# Output panel

<img width="1268" alt="Screenshot 2021-11-19 at 16 23 30" src="https://user-images.githubusercontent.com/5427083/142647353-ebc5ee07-3f2f-4d94-90d8-08301d16892b.png">

**Comment by witoldsz** [comment]
@Krzysztof-Cieslak the only log in the F# Output Panel is:
```
[Info  - 5:42:37 PM] Connection to server got closed. Server will restart.
[Info  - 5:42:37 PM] Connection to server got closed. Server will restart.
[Info  - 5:42:37 PM] Connection to server got closed. Server will restart.
[Info  - 5:42:38 PM] Connection to server got closed. Server will restart.
[Error - 5:42:38 PM] Connection to server got closed. Server will not be restarted.
```

**Comment by StailGot** [comment]
Another one. 

Debian 11,
λ: dotnet --list-sdks
6.0.100 [/usr/share/dotnet/sdk]

v5.8.1 works ok, all later fail to start with 
> Connection to server got closed. Server will not be restarted.

**Comment by baronfel** [comment]
Can you all try setting the 'FSharp.dotnetRoot' setting to the directory that contains your dotnet executable? Also please note that if you're using gitpod, there's an issue at the moment with their default environments that breaks most dotnet tooling.

**Comment by StailGot** [comment]
No gitpod used.
Tried set dotnetRoot to /usr/share/dotnet/sdk/6.0.100/ and /usr/share/dotnet. Nothing.

**Comment by OriginatorStar** [comment]
What I get from the F# output is a reference to a folder path that ionide seems to send, and it breaks it:
(I've already set FSharp.dotnetRoot)

`Unhandled exception: System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'c:\repos\fs\FirstIonideProject\dotnet' with working directory 'c:\repos\fs\FirstIonideProject'. The system cannot find the file specified.`

That dotnet folder is not created by default when scaffolding a F# project. To me ionide seems to be failing to look for the dotnet.exe ?

Give me some clues please on where should I look to correct this. I've checked my system path env var, and the first thing there is the path to dotnet sdk.

**Comment by witoldsz** [comment]
Today I have found the cause of all the troubles. I had `DOTNET_PATH` set in my `~/.profile` to some non-existing directory (I had dotnet installed manually on my previous PC and I've copied the .profile file over to the new one.

Now, the Ionide plugin works again. It's been a loooong while, because after the incident, I have switched to JetBrains Rider. It cannot handle `.fsx` files with `#r "nuget: …" dependencies, so I've tried the Ionide again and finally it works :)
